---
layout: page
category: cook
title: "Create a BLAST database"
order: 1
---

First you need to create a BLAST database for your genome or transcriptome. For your reference sequences in a FASTA file, use this command line:

```
makeblastdb -in <reference.fa> -dbtype nucl -parse_seqids -out <database_name> -title "Database title"
```

The ```-parse_seqids``` option is required to keep the original sequence identifiers. Otherwise makeblastdb will generate its own identifiers, ```-title``` is optional.

For more information on makeblastdb see [NCBI BLAST+ Command Line User Manual](https://www.ncbi.nlm.nih.gov/books/NBK279688/).

Magic-BLAST will work with a genome in a FASTA file, but will be very slow for anything larger than a bacterial genome, so we do not recommend it.

 
&nbsp;


#### Example

To create a BLAST database from the reference file ```my_reference.fa```

```
$ cat my_reference.fa
>sequence_1 Homo sapiens hemoglobin subunit alpha 2 (HBA2), mRNA
CATAAACCCTGGCGCGCTCGCGGGCCGGCACTCTTCTGGTCCCCACAGACTCAGAGAGAACCCACCATGG
TGCTGTCTCCTGCCGACAAGACCAACGTCAAGGCCGCCTGGGGTAAGGTCGGCGCGCACGCTGGCGAGTA
TGGTGCGGAGGCCCTGGAGAGGATGTTCCTGTCCTTCCCCACCACCAAGACCTACTTCCCGCACTTCGAC
CTGAGCCACGGCTCTGCCCAGGTTAAGGGCCACGGCAAGAAGGTGGCCGACGCGCTGACCAACGCCGTGG
CGCACGTGGACGACATGCCCAACGCGCTGTCCGCCCTGAGCGACCTGCACGCGCACAAGCTTCGGGTGGA
CCCGGTCAACTTCAAGCTCCTAAGCCACTGCCTGCTGGTGACCCTGGCCGCCCACCTCCCCGCCGAGTTC
ACCCCTGCGGTGCACGCCTCCCTGGACAAGTTCCTGGCTTCTGTGAGCACCGTGCTGACCTCCAAATACC
GTTAAGCTGGAGCCTCGGTAGCCGTTCCTCCTGCCCGCTGGGCCTCCCAACGGGCCCTCCTCCCCTCCTT
GCACCGGCCCTTCCTGGTCTTTGAATAAAGTCTGAGTGGGCAGCAAAAAAAAAAAAAAAAAA
>sequence_2 Homo sapiens alpha one globin (HBA1) mRNA, complete cds
CAGACTCAGAGAGAACCCACCATGGTGCTGTCTCCTGCCGACAAGACCAACGTCAAGGCCGCCTGGGGTA
AGGTCGGCGCGCACGCTGGCGAGTATGGTGCGGAGGCCCTGGAGAGGATGTTCCTGTCCTTCCCCACCAC
CAAGACCTACTTCCCGCACTTCGACCTGAGCCACGGCTCTGCCCAGGTTAAGGGCCACGGCAAGAAGGTG
GCCGACGCGCTGACCAACGCCGTGGCGCACGTGGACGACATGCCCAACGCGCTGTCCGCCCTGAGCGACC
TGCACGCGCACAAGCTTCGGGTGGACCCGGTCAACTTCAAGCTCCTAAGCCACTGCCTGCTGGTGACCCT
GGCCGCCCACCTCCCCGCCGAGTTCACCCCTGCGGTGCACGCCTCCCTGGACAAGTTCCTGGCTTCTGTG
AGCACCGTGCTGACCTCCAAATACCGTTAAGCTGGAGCCTCGGTGGCCATGCTTCTTGCCCCTTGGGC
```

use this command line

```
makeblastdb -in my_reference.fa -out my_reference -parse_seqids -dbtype nucl
```

Note that the word following '>' is a sequence identifier that will be used in Magic-BLAST reports. The identifier should be unique.


### Download a genome

There are several ways to download whole genomes, transcriptomes, or selected sequences from NCBI.

#### NCBI Datasets

You can search and download genome and transcript sequences from [NCBI Datasets Genome](https://www.ncbi.nlm.nih.gov/datasets/genome/) page. Search for an organism, select an assembly, and you will see download options.

For example, you can download the GRCh38 assembly of the human genome, from https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000001405.40/.

#### NCBI EDirect tools

To download human chromosome 1 using [NCBI EDirect tools](https://github.com/NCBI-Hackathons/EDirectCookbook) use:

```
search -db nucleotide -query NC_000001 | efetch -format fasta >NC_000001.fa
```

You can download full human genome or transcriptome from [NCBI human genome resources](https://www.ncbi.nlm.nih.gov/projects/genome/guide/human/index.shtml) or use [NCBI Genome search](https://www.ncbi.nlm.nih.gov/genome) for any organism.

For  example to download the latest human genome use:

```
wget ftp://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_genomic.fna.gz
gzip -d GRCh38_latest_genomic.fna.gz
makeblastdb -in GRCh38_latest_genomic.fna -out GRCh38 -parse_seqids -dbtype nucl
```


